src/docfx/build/xref/InternalXrefMapBuilder.cs

```diff
@@ -147,18 +147,25 @@ private static InternalXrefSpec[] AggregateXrefSpecs(Context context, string uid
                 return new InternalXrefSpec[] { orderedConflict.First() };
             }
 
-            // uid conflicts with overlapping monikers, drop the uid and log an error
+            // uid conflicts with overlapping monikers
+            // log an warning and take the first one order by the declaring file
             var conflictsWithMoniker = specsWithSameUid.Where(x => x.Monikers.Count > 0).ToArray();
             if (CheckOverlappingMonikers(specsWithSameUid, out var overlappingMonikers))
             {
                 context.ErrorLog.Write(Errors.MonikerOverlapping(overlappingMonikers));
-                return Array.Empty<InternalXrefSpec>();
+                return new InternalXrefSpec[] { conflictsWithMoniker.OrderBy(item => item.DeclairingFile).First() };
```

Comment:
We can simply return the first one of order list which contains all xrefSpec with this UID no matter it has moniker or not by Declaring File here?

Reply:
sounds good, this could simplify the resolving logic as well